emcc.py

```diff
@@ -1849,6 +1849,12 @@ def check_memory_setting(setting):
        sanitize:
       shared.Settings.EXPORTED_FUNCTIONS += ['_malloc', '_free']
 
+    if not shared.Settings.DISABLE_EXCEPTION_CATCHING:
+      # If not for LTO builds, we could handle these by adding deps_info.py
+      # entries for __cxa_find_matching_catch_* functions.  However, under
+      # LTO these symbols don't exist prior the linking.
+      shared.Settings.EXPORTED_FUNCTIONS += ['___cxa_is_pointer_type', '___cxa_can_catch']
```

Comment:
Can we add this just for LTO builds? Or maybe that's less good because of added complexity.

What is the code size cost to having exceptions enabled but not needing these methods, and getting them anyhow?

Reply:
I don't think its worth the complexity.   This would mean that the developer asked for exceptions but never tried to catch one.. which would be odd.  Plus since these are native functions I'm pretty sure metadce can remove then right?

Reply:
Yes, they are native functions, but I'm not sure if they end up pulling in static data or function pointers which metadce can't remove.

Reply:
If you enable exceptions I think this is reasonable expectation that you include these functions.   How many programs with exceptions throwing enabled are going to be able to realistically build without these?

Reply:
Do you want me to investigate:

1. the non-metadce-able cost of including these when they are not needed?  
2. is it even possible to not include these in an exception-used codebase?

if the answer to 1 is zero or the answer to 2 is no then I this change is fine, right?      I think its fine anyway :)

Reply:
I think you're right. The risk to code size seems very low here. If we had exceptions on by default things would be different, but this is a benefit of us having them off normally.